src/module.c

```diff
@@ -10564,25 +10568,38 @@ int moduleFreeCommand(struct RedisModule *module, struct redisCommand *cmd) {
         cmd->latency_histogram = NULL;
     }
     zfree(cmd->args);
-    zfree(cp);
+    zfree((void*)(unsigned long)cmd->getkeys_proc);
```

Comment:
What is getkeys_proc doing here? I don't understand this.

Reply:
Since `getkeys_proc` is actually `RedisModuleCommand` when cmd is a module command, it seems like I should use union, but I don't want to change it too much.

Reply:
Oh, I see now that it was used like this in module.c already. I guess it is fine to keep it then, but it is confusing without a union or comments.

@oranagra did you review this when it was added?

Reply:
Like below suggestion it is at least visible that it's a RedisModuleCommand. Also, these lines will not appear to be changed in this PR. Only other lines between them are deleted.

```suggestion
    RedisModuleCommand *cp = (void*)(unsigned long)cmd->getkeys_proc;
    zfree(cp);
```

Reply:
At least it appears like less changes. :-) :+1: 

<details>
<summary>Diff screenshot</summary>

![image](https://user-images.githubusercontent.com/273886/153552703-99da63f4-1d28-43d3-8f46-1f29a001eb9c.png)
</details>

Reply:
> @oranagra did you review this when it was added?

you mean the abuse of `cmd->getkeys_proc` to save the module details?
no i did not review it.. it's part of the original implementation of the module system in redis 4.0.
not sure why he did that other than save 8 bytes in the command struct (doesn't seem like a real concern).
and it did cause some bugs in the past.

Reply:
Ah, right, it's that old.